.github/workflows/benchmark-action.yml

```diff
@@ -0,0 +1,112 @@
+name: Run performance benchmarks
+on:
+  workflow_dispatch: # Allows manual triggering of workflow
+  push:
+    branches:
+      - main
+    paths:
+      - src/client/Microsoft.Identity.Client/**/*.cs
+  #pull_request: # Will re-enable once the code to not push to gh-pages is fixed
+  #  types: [opened, reopened, synchronize, ready_for_review]
+  #  paths:
+  #    - src/client/Microsoft.Identity.Client/**/*.cs
+
+permissions:
+  # Deployments permission to deploy GitHub pages website
+  deployments: write
+  # Contents permission to update benchmark contents in gh-pages branch
+  contents: write
+
+jobs:
+  benchmark:
+    name: Run performance benchmarks
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-dotnet@v3
+        with:
+          dotnet-version: '3.1.426'
+      - name: Run benchmark
+        run: cd tests/Microsoft.Identity.Test.Performance && dotnet run -c release -f netcoreapp3.1 --exporters json 
+
+      - name: Download previous benchmark data
+        uses: actions/cache@v3
+        with:
+          path: ./cache
+          key: ${{ runner.os }}-benchmark
+
+      - name: Store AcquireTokenNoCacheTests result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: AcquireTokenNoCache
+          tool: 'benchmarkdotnet'
+          output-file-path: tests/Microsoft.Identity.Test.Performance/BenchmarkDotNet.Artifacts/results/Microsoft.Identity.Test.Performance.AcquireTokenNoCacheTests-report-full-compressed.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
+          #external-data-json-path: ./cache/benchmark-data.json
+          # Show alert with commit comment on detecting possible performance regression
+          alert-threshold: '140%'
+          comment-on-alert: true
+          fail-on-alert: true
+          alert-comment-cc-users: '@pmaytak'
+          summary-always: true
+          gh-pages-branch: gh-pages
+          benchmark-data-dir-path: benchmarks
+
+      - name: Store AcquireTokenForClientCacheTests result
+        uses: benchmark-action/github-action-benchmark@v1
+        with:
+          name: AcquireTokenForClientWithCache
+          tool: 'benchmarkdotnet'
+          output-file-path: tests/Microsoft.Identity.Test.Performance/BenchmarkDotNet.Artifacts/results/Microsoft.Identity.Test.Performance.AcquireTokenForClientCacheTests-report-full-compressed.json
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          auto-push: true
+          #external-data-json-path: ./cache/benchmark-data.json
+          # Show alert with commit comment on detecting possible performance regression
+          alert-threshold: '140%'
+          comment-on-alert: true
+          fail-on-alert: true
+          alert-comment-cc-users: '@pmaytak'
+          summary-always: true
+          gh-pages-branch: gh-pages
+          benchmark-data-dir-path: benchmarks
```

Comment:
In the graph page, can you change the name of each test to something more meaningful (currently it's the test FQN)?

It would also be interesting to combine some of those graphs toghether, for example "GetAccounts" + "ATS" + "AcquireTokenByCode" or "EnableCacheSync = True / False"

Reply:
I don't think so. We specify a user-friendly description, but seems like it's not picked up.